app/src/main/java/br/com/catbag/gifreduxsample/helpers/AppStateHelper.java

```diff
@@ -0,0 +1,46 @@
+package br.com.catbag.gifreduxsample.helpers;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import br.com.catbag.gifreduxsample.models.AppState;
+import br.com.catbag.gifreduxsample.models.Gif;
+
+/**
+ * Created by felipe on 21/10/16.
+ */
+
+public final class AppStateHelper {
+
+    private AppStateHelper() { }
+
+    public static Gif getFirstGif(AppState state) {
+        if (state.getGifs().size() <= 0) return null;
+        return toGifList(state.getGifs()).get(0);
+    }
+
+    public static Gif getLastGif(AppState state) {
+        if (state.getGifs().size() <= 0) return null;
+        return toGifList(state.getGifs()).get(state.getGifs().size());
+    }
+
+    public static Gif getGifStateByUuid(String uuid, AppState state) {
+        return state.getGifs().get(uuid);
+    }
+
+    private static List<Gif> toGifList(Map map) {
+        return new ArrayList<>(map.values());
+    }
+
+    public static List<Gif> getListGifs(AppState state) {
+        return toGifList(state.getGifs());
+    }
+
+    public static Map<String, Gif> toMap(List<Gif> gifs) {
+        Map<String, Gif> map = new HashMap<>();
+        for (Gif gif : gifs) map.put(gif.getUuid(), gif);
+        return map;
+    }
+}
```

Comment:
I don't have a strong opinion about this, maybe this object being simple and clear, with only immutable methods (that have yours own complexity) will be more easy to maintain and understand app model. Thinking that this objects will be more representative of all app information. Need a live discussion. Opening a issue.